controller/nodegroup_test.go

```diff
@@ -0,0 +1,131 @@
+package controller
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	v1 "github.com/rancher/eks-operator/pkg/apis/eks.cattle.io/v1"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetNodegroupConfigUpdate(t *testing.T) {
+	type nodegroupUpdateTestCase struct {
+		clusterName string
+		ng1 v1.NodeGroup
+		ng2 v1.NodeGroup
+		expectedNgUpdateInput eks.UpdateNodegroupConfigInput
+		expectedNgNeedsUpdate bool
+	}
```

Comment:
nit: its nice to have a basic case title with each of them. Like "delete and update of old labels".

Reply:
Added.

Reply:
Oh I actually meant in the struct, so the test output would include the title. But we can save that for next time, as long as the case summary exists alongside the test.